internal/generate/scaffold.go

```diff
@@ -0,0 +1,95 @@
+package generate
+
+import (
+	"bytes"
+	"embed"
+	"fmt"
+	"io/fs"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/pkg/errors"
+)
+
+//go:embed scaffold
+var scaffoldFS embed.FS
+
+type ScaffoldData struct {
+	AppName        string
+	DeploymentName string
+	Description    string
+	Environment    string
+	Namespace      string
+}
+
+func AppScaffold(appName string, environments []string, outputPath string) error {
+	for _, env := range environments {
+		files, err := generateAppScaffoldEnvironmentFiles(appName, env)
+		if err != nil {
+			return err
+		}
+
+		if err := files.Save(filepath.Join(outputPath, env)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func generateAppScaffoldEnvironmentFiles(appName, environment string) (*Files, error) {
+	tpl, err := template.New("tpl").ParseFS(scaffoldFS, "scaffold/*.yaml", "scaffold/**/*.yaml")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to create template")
+	}
+
+	data := ScaffoldData{
+		AppName:        appName,
+		DeploymentName: fmt.Sprintf("%s-environment-%s", environment, appName),
+		Description:    fmt.Sprintf("%s example application", appName),
+		Environment:    environment,
+		Namespace:      environment,
+	}
+
+	files := &Files{}
+	err = fs.WalkDir(scaffoldFS, ".", func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return errors.Wrap(err, "error walking directory")
+		}
+
+		if d.IsDir() {
+			return nil
+		}
+
+		// Get the file name without the root path
+		s := strings.Split(path, string(filepath.Separator))
+		file := strings.Join(s[1:], string(filepath.Separator))
```

Comment:
Might be better to use something like https://pkg.go.dev/path/filepath#Base

Reply:
They don't do quite the same thing. This takes the string `scaffold/{{ .AppName }}/values.yaml` and returns `{{ .AppName }}/values.yaml`.

For the same string, `filepath.Base` would return `values.yaml`.

I tried in vain to find a `filepath` function which removes just the root directory - if you know of something that does this, please share.

Reply:
Can you do a bounds check if none of the stdlib can handle it? That `s[1:]` could panic if the path is not big enough to be separated.

You can prove this with a unit test too btw.

Reply:
It's always going to be at least one because of how the `embed` works, but I've reworked it